src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,178 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private transient MyLinkedList.Node<T> first;
+    private transient MyLinkedList.Node<T> last;
+    private transient int size = 0;
+
     @Override
     public void add(T value) {
+        Node<T> val = new Node<>(null, value, null);
+        if (size == 0) {
+            first = last = val;
+        } else {
+            swapLast(val);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> givenNode = new Node<>(null, value, null);
+        Node<T> current = node(index);
+
+        if (index == 0) {
+            swapFirst(givenNode);
+        } else if (index == size) {
+            swapLast(givenNode);
```

Comment:
In the `add` method, when `index == size`, the `swapLast` method is called, which is correct. However, ensure that the `givenNode` is correctly linked to the previous last node, as `swapLast` should handle this correctly.